job-service/src/main/java/com/hpe/caf/services/job/api/DatabaseHelper.java

```diff
@@ -221,19 +242,21 @@ public Job getJob(final String partitionId, String jobId) throws Exception {
                     }
                 }
             }
-        } catch (SQLException se) {
+        } catch (final SQLException se) {
             //  Determine source of SQL exception and throw appropriate error.
-            String sqlState = se.getSQLState();
-
-            switch (sqlState) {
-                case "02000":
-                    //  Job id has not been provided.
-                    throw new BadRequestException(se.getMessage());
-                case "P0002":
-                    //  No data found for the specified job id.
-                    throw new NotFoundException(se.getMessage());
-                default:
-                    throw se;
+            final String sqlState = se.getSQLState();
+
+            if (sqlState.equals(POSTGRES_NO_DATA_ERROR_CODE)) {
+                //  Job id has not been provided.
+                throw new BadRequestException(se.getMessage());
+            } else if (sqlState.equals(POSTGRES_NO_DATA_FOUND_ERROR_CODE)) {
+                //  No data found for the specified job id.
+                throw new NotFoundException(se.getMessage());
+            } else if (sqlState.startsWith(POSTGRES_CONNECTION_EXCEPTION_ERROR_CODE_PREFIX)) {
+                // Connection exception.
+                throw new ServiceUnavailableException(se.getMessage(), se);
```

Comment:
I see this pattern repeating over and over.  Would it be a good idea to have a `mapSqlException()` utility function which takes the `SQLException` and returns the `Exception` to be thrown?

Reply:
@dermot-hardy @kleyton-ohare Just be aware that all the catch blocks are not the same, so changing them to all perform the same exception processing is changing the existing behaviour now.

For example, in the original code for the `canJobBeProgressed` function, if the error code is P0002 the method returns false:

```
        } catch (SQLException e) {
            final String sqlState = e.getSQLState();
            if (sqlState.equals("P0002")) {
                // job missing - return false
            } else {
                throw e;
            }
        }
        return canBeProgressed;
```

Now the method will throw a NotFoundException instead of returning false.

```
 else if (sqlState.equals(POSTGRES_NO_DATA_FOUND_ERROR_CODE)) {
            //  No data found for the specified job id.
            return new NotFoundException(se.getMessage());
        }
```

There are other examples as well - methods that may have thrown an SQLException previously may now throw one of the other exceptions, `NotFoundException`, `BadRequestException`.

Reply:
@kleyton-ohare I had a quick chat with Dermot regarding this. 

It may be useful to look through the code history to try and determine why some of the catch blocks are different, i.e. if it's intentional, or maybe if some catch blocks were missed when updating others etc.

If you find that the existing code is correct and we should maintain the same behaviour, then instead of having one mapException method, we could maybe have 2 or 3 mapException methods to encapsulate the common behaviour.